### PR TITLE
Add slider for zap-outs, prevent swaps when price impact is too high

### PR DESCRIPTION
**Pull request by jophish**
This PR adds a slider interface for zap-outs. I figured it would be good to prevent zap-outs when price impact becomes too high, since currently it's possible to zap in/out with very high price impact by accident. Incidentally, these changes also prevent zap-ins when price impact is too high as well.

2022-02-28-191315_1025x1211_scrot

2022-02-28-191358_1035x784_scrot

### CODE REVIEW

File: src/components/compound/ConfirmAddCompoundModalBottom.tsx [hunk 1]
```diff
@@ -1,11 +1,10 @@
-import { Fraction, Percent, Token, TokenAmount } from '@ubeswap/sdk'
+import { Token, TokenAmount } from '@ubeswap/sdk'
```
Comment (jophish) [inline src/components/compound/ConfirmAddCompoundModalBottom.tsx:1]: The changes to this file and `ConfirmWithdrawCompoundModalBottom.tsx` were done automatically when I ran the linter.

File: src/pages/App.tsx [hunk 2]
```diff
@@ -89,7 +89,7 @@ export default function App() {
       <Route component={DarkModeQueryParamReader} />
       <AppWrapper>
         <ConstructionBanner>
-          Warning: Revo's smart contracts are pending full audits. Please use with caution!
+          Warning: Revo&apos;s smart contracts are pending full audits. Please use with caution!
```
Comment (jophish) [inline src/pages/App.tsx:92]: Linter gave an error (not just a warning) about this; went ahead and fixed it here.

File: src/utils/index.ts [hunk 3]
```diff
@@ -9,7 +9,6 @@ import { IUniswapV2Router02, UbeswapMoolaRouter } from 'generated/index'
 import { ROUTER_ADDRESS, UBESWAP_MOOLA_ROUTER_ADDRESS } from '../constants'
 import IUniswapV2Router02ABI from '../constants/abis/IUniswapV2Router02.json'
 import UbeswapMoolaRouterABI from '../constants/abis/UbeswapMoolaRouter.json'
```
Comment (jophish) [inline src/utils/index.ts:11]: Again, this was the linter.

File: src/components/earn/PoolCard.tsx [hunk 4]
```diff
@@ -113,7 +127,14 @@ export const PoolCard: React.FC<Props> = ({ compoundBotSummary }: Props) => {
...
   const { address } = useContractKit()
   const dispatch = useDispatch()
+
   const { v2Trade: trade, inputError: swapInputError } = useDerivedSwapInfo()
+  const { priceImpactWithoutFee } = trade ? computeTradePriceBreakdown(trade) : {}
+  const isPriceImpactTooHigh = priceImpactWithoutFee
+    ? priceImpactWithoutFee.lessThan(BLOCKED_PRICE_IMPACT_NON_EXPERT)
+      ? false
+      : true
+    : false
```
Comment (cajubelt) [inline src/components/earn/PoolCard.tsx]:
```suggestion
  const isPriceImpactTooHigh = priceImpactWithoutFee?.greaterThan(BLOCKED_PRICE_IMPACT_NON_EXPERT)
```
The double ternary is very confusing. I think this is logically equivalent.

Comment (cajubelt) [reply src/components/earn/PoolCard.tsx]: you could add a `!!` prefix if you are worried about it being undefined

**Review by cajubelt — COMMENTED** [review]
very cool! Just skimmed and commented on an area that I think could be clearer

File: src/components/earn/PoolCard.tsx (hunk 4 above)
Comment (jophish) [reply src/components/earn/PoolCard.tsx]: If it's undefined, we'll get a TypeError when attempting to call `lessThan` or `greaterThan` on it though, right?

Comment (jophish) [reply src/components/earn/PoolCard.tsx]: I definitely agree that the double ternary is confusing, though.

Comment (cajubelt) [reply src/components/earn/PoolCard.tsx]: I think the question mark before the period will prevent the TypeError

Comment (jophish) [reply src/components/earn/PoolCard.tsx]: oh you're totally right, ty!

**Pull request merged**